robusta_krr/main.py

```diff
@@ -26,281 +25,272 @@
+        rich_help_panel="Prometheus EKS Settings",
+    ),
+    eks_access_key: Optional[str] = typer.Option(
+        None,
+        "--eks-access-key",
+        help="Sets the access key for eks prometheus connection.",
+        rich_help_panel="Prometheus EKS Settings",
+    ),
+    eks_secret_key: Optional[str] = typer.Option(
+        None,
+        "--eks-secret-key",
+        help="Sets the secret key for eks prometheus connection.",
+        rich_help_panel="Prometheus EKS Settings",
+    ),
+    eks_service_name: Optional[str] = typer.Option(
+        "aps",
+        "--eks-service-name",
+        help="Sets the service name for eks prometheus connection.",
+        rich_help_panel="Prometheus EKS Settings",
+    ),
+    eks_managed_prom_region: Optional[str] = typer.Option(
+        None,
+        "--eks-managed-prom-region",
+        help="Sets the region for eks prometheus connection.",
+        rich_help_panel="Prometheus EKS Settings",
+    ),
+    coralogix_token: Optional[str] = typer.Option(
+        None,
+        "--coralogix-token",
+        help="Adds the token needed to query Coralogix managed prometheus.",
+        rich_help_panel="Prometheus Coralogix Settings",
+    ),
+    openshift: bool = typer.Option(
+        False,
+        "--openshift",
+        help="Used when running by Robusta inside an OpenShift cluster.",
+        rich_help_panel="Prometheus Openshift Settings",
+        hidden=True,
+    ),
+    max_workers: int = typer.Option(
+        10,
+        "--max-workers",
+        "-w",
+        help="Max workers to use for async requests.",
+        rich_help_panel="Threading Settings",
+    ),
+    format: str = typer.Option(
+        "table",
+        "--formatter",
+        "-f",
+        help=f"Output formatter ({', '.join(formatters.list_available())})",
+        rich_help_panel="Logging Settings",
+    ),
+    verbose: bool = typer.Option(
+        False, "--verbose", "-v", help="Enable verbose mode", rich_help_panel="Logging Settings"
+    ),
+    quiet: bool = typer.Option(
+        False, "--quiet", "-q", help="Enable quiet mode", rich_help_panel="Logging Settings"
+    ),
+    log_to_stderr: bool = typer.Option(
+        False, "--logtostderr", help="Pass logs to stderr", rich_help_panel="Logging Settings"
+    ),
+    width: Optional[int] = typer.Option(
+        None,
+        "--width",
+        help="Width of the output. Will use console width by default.",
+        rich_help_panel="Logging Settings",
+    ),
+    file_output: Optional[str] = typer.Option(
+        None, "--fileoutput", help="Print the output to a file", rich_help_panel="Output Settings"
+    ),
+    slack_output: Optional[str] = typer.Option(
+        None,
+        "--slackoutput",
+        help="Send to output to a slack channel, must have SLACK_BOT_TOKEN",
+        rich_help_panel="Output Settings",
+    ),
+    cpu_min_value: int = __pydantic_field_to_typer_option(
```

Comment:
Also here, it is now required to add those fields here manually. For our own codebase it is ok, but it will be much more code for someone to implement a strategy himself

Note: there were some users in discussions that told that they are making their own strategies, so I feel like such a big added boilerplate is bad

Reply:
I understand, but we're going to optimize for simplicity and getting more PRs, even at the expense of introducing boilerplate.